### DISCUSSION TITLE
Feature: Charge/Wavefunction Extrapolation with External Interfaces

### DISCUSSION THREAD

**Discussion by QuantumMisaka** [deepmodeling/abacus-develop#5290] [General]
### Background

#3264 is a start for external interfaces to use electronic structure information from ABACUS to do next ionic step, but only read-in the charge and wavefunction file for initialization of next ionic step is not enough, extrapolation for wavefunction & charge is needed when using external interface like ASE to do optimization, NEB or other jobs.

ABACUS have charge/wavefunction extrapolation now, the only thing is to use it.

### Describe the solution you'd like

Modification of wavefunction and charge file read-in part and charge initialization part, which allows ABACUS to do extrapolation with read-in charge and wavefunction file from last ionic step and read-in STRU for next ionic step. 

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Additional Context

I'd like to make it my first contribution for ABACUS source code with some help.

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by WHUweiqingzhou** [comment]
@QuantumMisaka , May I ask your further plan for this issue?
